DXRMapFixups/DeusEx/Classes/DXRFixupVandenberg.uc

```diff
@@ -704,20 +725,69 @@ function TimerMapFixes()
         break;
 
     case "14_Oceanlab_silo":
-        prevMapsDone = dxr.flagbase.GetBool('Heliosborn') &&  //Finished Vandenberg, mission 12
-            dxr.flagbase.GetBool('schematic_downloaded'); //Finished Ocean Lab, mission 14,
-        prevMapsDone = prevMapsDone || !#defined(injections) || dxr.flags.settings.goals<=0;
+        // by design, no infolink plays after killing Howard Strong if the missile hasn't been redirected
+        _MissileLaunchedChecks();
+        _CheckMoreSiloInfolinks();
+        break;
+    }
+}
 
-        if(prevMapsDone
-           && dxr.flagbase.GetBool('missile_launched') //Redirected the missile in Silo, mission 14
-           && !dxr.flagbase.GetBool('MS_HowardStrongUnhidden'))
-        {
-            foreach AllActors(class'#var(prefix)HowardStrong', hs) {
-                hs.EnterWorld();
+function private _MissileLaunchedChecks() {
+    local #var(prefix)HowardStrong hs;
+    local bool strongAlive;
+    local BlackHelicopter chopper;
+
+    if (dxr.flagbase.GetBool('DXR_SiloEscapeHelicopterUnhidden') || !dxr.flagbase.GetBool('missile_launched')) {
+        return;
+    }
+
+    foreach AllActors(class'HowardStrong', hs) {
+        strongAlive = True;
+        break;
+    }
+
+    if (strongAlive == False) {
+        // both goals completed just now
+        foreach AllActors(class'BlackHelicopter', chopper, 'BlackHelicopter') {
+            chopper.EnterWorld();
+            break;
+        }
+        dxr.flagbase.SetBool('DXR_SiloEscapeHelicopterUnhidden', True,, 15);
+    } else {
+        // only computer goal completed
+        player().StartDataLinkTransmission("DL_Savage3");
+    }
+}
+
+function private _CheckMoreSiloInfolinks() {
```

Comment:
This function name should be more descriptive to what it's actually doing.  SiloExitInfolinkModification()?

Not sure if declaring the functions as private is really necessary (if it has any benefits whatsoever, tbh)

Reply:
Yeah, I couldn't think of a good name so I just went with Die's suggestion. Logically it's tied to _MissileLaunchedChecks still.

As for making them private, that's really to drive home that they are really just pieces of TimerMapFixes and useless to anything else. The benefit is mostly for the reader. But it would be rather poor style to expose them either way.